mods/env_sounds/init.lua

```diff
@@ -0,0 +1,97 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:get_pos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
```

Comment:
There's much duplicated code. What about something like this:

```lua
local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
local water_names = {"default:water_flowing", "default:river_water_flowing"}
if river_source_sounds
	table.insert(water_names, "default:river_water_source")
end

--...

	local wpos, _ = minetest.find_nodes_in_area(areamin, areamax, water_names)
```

Reply:
Will do thanks.

Reply:
You could've cached the `water_nodes` table on mod load as `river_source_sounds` doesn't change.